pandas/io/formats/style.py

```diff
@@ -171,6 +171,13 @@ def __init__(
         self.cell_ids = cell_ids
         self.na_rep = na_rep
 
+        self.tooltip_styles: List[Dict[str, object]] = []
```

Comment:
does it make sense to make this a named tuple? (or a class/dataclass) for clarity here?

Reply:
I have re-written this PR with a separate class approach, to aid with other comments also...